main/utils/cnfs.c

```diff
@@ -6,11 +6,51 @@
 #include <string.h>
 
 #include <esp_err.h>
-#include "../../tools/cnfs/image.h"
-#include "../../tools/cnfs/image.c"
 #include <esp_log.h>
 #include <esp_heap_caps.h>
-#include "hdw-spiffs.h"
+#include "cnfs.h"
+
+#if EMULATOR
```

Comment:
Is `EMULATOR` actually defined in the build system? I tried really hard to _not_ have any build-specific preprocessor logic in the code. The emulator _should_ read out of `cnfs_image.c` just like the firmware now, right? I want both builds to be as close as possible, and only mock hardware or IDF features.